core/src/main/java/org/elasticsearch/action/admin/indices/shards/IndicesShardStoresResponse.java

```diff
@@ -198,7 +198,7 @@ public void writeTo(StreamOutput out) throws IOException {
         @Override
         public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
             node.toXContent(builder, params);
-            builder.field(Fields.VERSION, version);
+            builder.field(Fields.VERSION, legacyVersion);
```

Comment:
depends (which is why I asked). If it's about API bwc I think we should break it and be clear about the impact of the version. Which is also makes me think we should not render the field if we don't have a version (because we removed it)